src/Ng/Jobs/BackupV3Storage.cs

```diff
@@ -0,0 +1,236 @@
+            string logFile = $"{destContainer}.log";
+            string azCopyLogPath = Path.Combine(azCopyTempFolder, logFile);
+            RefreshLogData(azCopyTempFolder, azCopyLogPath);
+
+            if (await TryCreateDestinationContainerAsync(destContainer, _destStorageAccountName, _destStorageKeyValue))
+            {
+                string arguments = $"/Source:https://{sourceAccountName}.blob.core.windows.net/{container}/ " +
+                                   $"/Dest:https://{_destStorageAccountName}.blob.core.windows.net/{destContainer}/ " +
+                                   $"/SourceKey:{sourceAccountKey} /DestKey:{_destStorageKeyValue} " +
+                                   $"/S /Z:{azCopyTempFolder} /V:{azCopyLogPath}";
+
+                string argumentsLog = $"/Source:{azCopyTempFolder} /Dest:https://{_destStorageAccountName}.blob.core.windows.net/logs" +
+                                      $" /DestKey:{_destStorageKeyValue} /destType:blob /Pattern:{logFile}";
+
+                try
+                {
+                    ProcessStartInfo copyToAzureProc = new ProcessStartInfo();
+                    copyToAzureProc.FileName = @"cmd.exe";
+                    copyToAzureProc.Arguments = $"/c {AzCopyPath} {arguments}";
+                    copyToAzureProc.CreateNoWindow = false;
+                    copyToAzureProc.UseShellExecute = false;
+#if Debug
+                copyToAzureProc.RedirectStandardOutput = true;
+#endif
+
+                    Logger.LogInformation($"StartContainerCopy:{container}");
+                    sw.Start();
+                    Process p = Process.Start(copyToAzureProc);
+                    p.WaitForExit();
+#if Debug
+                string result = p.StandardOutput.ReadToEnd();
+                Console.Write(result);
+#endif
+                    sw.Stop();
+                    int exitCode = p.ExitCode;
+                    Logger.LogInformation($"EndContainerCopy:{container}:{exitCode}:{sw.ElapsedMilliseconds}");
+                    p.Close();
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError($"BackupV3Storage:Exception on backup save.{ex.ToString()}.");
+                    return false;
+                }
+                try
+                {
+                    ProcessStartInfo copyToAzureProcLog = new ProcessStartInfo();
+                    copyToAzureProcLog.FileName = @"cmd.exe";
+                    copyToAzureProcLog.Arguments = $"/c {AzCopyPath} {argumentsLog}";
+                    copyToAzureProcLog.CreateNoWindow = false;
+                    copyToAzureProcLog.UseShellExecute = false;
+                    Process pLog = Process.Start(copyToAzureProcLog);
+                    pLog.WaitForExit();
+                    pLog.Close();
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError($"BackupV3Storage:Exception on log save.{ex.ToString()}.");
```

Comment:
The AzCopy logs are saved to the Azure storage. They should contain all the info needed for detailed debugging.
PR: https://github.com/NuGet/NuGet.Jobs/pull/151